cli/src/main/java/org/jboss/as/cli/handlers/ReadAttributeHandler.java

```diff
@@ -116,8 +116,12 @@ public boolean canAppearNext(CommandContext ctx) throws CommandFormatException {
                     op.get(Util.ADDRESS).setEmptyList();
                 } else {
                     final ModelNode addrNode = op.get(Util.ADDRESS);
-                    for(OperationRequestAddress.Node node : address) {
-                        addrNode.add(node.getType(), node.getName());
+                    for (OperationRequestAddress.Node node : address) {
+                        if (!(node.getName() == null)) {
```

Comment:
Thank-you for the change, I would suggest a tiny change to first deal with the null case if (node.getName() == null), it simplify a bit code reading.